src/main/scala/br/unb/cic/oberon/ast/OberonModule.scala

```diff
@@ -172,6 +172,18 @@ case class AndExpression(left: Expression, right: Expression) extends Expression
 /* Statements */
 trait Statement {
   def accept(v: OberonVisitor): v.T = v.visit(this)
+
+  val _id: Int = Statement.getNextId()
+  override def hashCode(): Int = _id
+}
+
+object Statement {
+  private var nextId = 0
+  def getNextId() = {
+    val returnId = nextId;
+    nextId += 1;
+    returnId
+  }
 }
```

Comment:
I liked this idea @delanoerhardt . Question: adding this `id` field did not break pattern matching, right?

Reply:
It didn't, none of the other test suites were affected by this.